stickyheaders/src/main/java/com/brandongogetap/stickyheaders/StickyHeaderPositioner.java

```diff
@@ -233,6 +242,7 @@ void attachHeader(RecyclerView.ViewHolder viewHolder, int headerPosition) {
         if (checkMargins) {
             updateLayoutParams(currentHeader);
         }
+        recyclerView.getViewTreeObserver().addOnGlobalLayoutListener(visibilityObserver);
```

Comment:
I think we'd be safe since everything is on the main thread, but just for logical consistency can you add the listener before the call to `addView`?